src/components/tl/ucp/scatter/scatter_knomial.c

```diff
@@ -165,8 +165,8 @@ void ucc_tl_ucp_scatter_knomial_progress(ucc_coll_task_t *coll_task)
                                      &offset, &local_seg_count);
     if (offset != 0) {
         status = ucc_mc_memcpy(PTR_OFFSET(args->dst.info.buffer, offset),
-                               PTR_OFFSET(rbuf, task->scatter_kn.send_offset),
-                               local_seg_count * dt_size, mem_type, mem_type);
+                               rbuf, task->scatter_kn.recv_size, mem_type,
```

Comment:
why copying from rbuf without send_offset?

Reply:
we want to copy everything that we receive so far to have less number of communications in allgather step after it